logback-contrib/json/core/src/main/java/ch/qos/logback/contrib/json/JsonLayoutBase.java

```diff
@@ -71,15 +72,17 @@ protected String formatTimestamp(long timestamp) {
         if (this.timestampFormat == null || timestamp < 0) {
             return String.valueOf(timestamp);
         }
-        Date date = new Date(timestamp);
-        DateFormat format = createDateFormat(this.timestampFormat);
+        Instant instant = Instant.ofEpochMilli(timestamp);
 
         if (this.timestampFormatTimezoneId != null) {
-            TimeZone tz = TimeZone.getTimeZone(this.timestampFormatTimezoneId);
-            format.setTimeZone(tz);
+            dateTimeFormatter = DateTimeFormatter.ofPattern(this.timestampFormat)
+                    .withZone(ZoneId.of(this.timestampFormatTimezoneId));
+        } else {
+            dateTimeFormatter = DateTimeFormatter.ofPattern(this.timestampFormat)
+                    .withZone(ZoneId.systemDefault());
         }
```

Comment:
The key idea is that we can move dateTimeFormatter into the constructor so that it gets reused